sdk/storage/azure-storage-queue/azure/storage/queue/_encryption.py

```diff
@@ -396,19 +408,22 @@ def adjust_blob_size_for_encryption(size: int, encryption_data: Optional[_Encryp
     :rtype: int
     """
     if is_encryption_v2(encryption_data):
-        nonce_length = encryption_data.encrypted_region_info.nonce_length
-        data_length = encryption_data.encrypted_region_info.data_length
-        tag_length = encryption_data.encrypted_region_info.tag_length
-        region_length = nonce_length + data_length + tag_length
-
-        num_regions = math.ceil(size / region_length)
-        metadata_size = num_regions * (nonce_length + tag_length)
-        return size - metadata_size
+        if encryption_data is not None:
+            if encryption_data.encrypted_region_info is not None:
```

Comment:
I can give it another shot, iirc the reason why these are split is because MyPy couldn't infer from compound if statements. If that is the case, would we prefer to keep it with these cascading `if` statements, or just throw ignores on the original?

Reply:
1. Probably get rid of `Optional[` for `encryption_data`
2. May just have to use ignore here since we have this whole if this param then this param